crates/cheatnet/src/runtime_extensions/forge_runtime_extension/mod.rs

```diff
@@ -217,12 +219,36 @@ impl<'a> ExtensionLogic for ForgeExtension<'a> {
                 let contract = input_reader.read()?;
                 let class = input_reader.read()?;
 
-                extended_runtime
+                let res = if extended_runtime
                     .extended_runtime
-                    .extension
-                    .cheatnet_state
-                    .replace_class_for_contract(contract, class);
-                Ok(CheatcodeHandlingResult::Handled(vec![]))
+                    .extended_runtime
+                    .hint_handler
+                    .state
+                    .get_class_hash_at(contract)
+                    .unwrap()
+                    == ClassHash::default()
+                {
+                    Err(ReplaceBytecodeError::ContractNotDeployed)
+                } else if matches!(
+                    extended_runtime
+                        .extended_runtime
+                        .extended_runtime
+                        .hint_handler
+                        .state
+                        .get_compiled_contract_class(class),
+                    Err(StateError::UndeclaredClassHash(_))
+                ) {
```

Comment:
In theory a state read error here can happen (e.g. internet connection failure during fork testing). We should bubble the `StateReadError`

Reply:
[Bubble state errors](https://github.com/foundry-rs/starknet-foundry/pull/2106/commits/e4483d5b8973bd04af563acdb056a8a48e9e29a4)